### PR TITLE
Reorganize plugin package into sub packages.

### PR DESCRIPTION
**Pull request by anusha-ragunathan**
Split plugin package into `store` and `v2/plugin`. Now the functionality
is clearly delineated:
- Manager: Manages the global state of the plugin sub-system.
- PluginStore: Manages a collection of plugins (in memory and on-disk)
- Plugin: Manages the single plugin unit.

This also facilitates splitting the global PluginManager lock into:
- PluginManager lock to protect global states.
- PluginStore lock to protect store states.
- Plugin lock to protect individual plugin states.

Importing "github.com/docker/docker/plugin/store" will provide access
to plugins and has lesser dependencies when compared to importing the
original monolithic `plugin package`.

Signed-off-by: Anusha Ragunathan anusha@docker.com

### CODE REVIEW

**Comment by icecrime** [comment]
Assigning @tiborvass and @vieux for review: PTAL!

**Comment by vieux** [comment]
LGTM, tested 👍

File: plugin/manager.go [hunk 1]
```diff
@@ -34,80 +29,22 @@ var (
...
 
-type pluginMap map[string]*plugin
 type eventLogger func(id, name, action string)
 
 // Manager controls the plugin subsystem.
 type Manager struct {
 	sync.RWMutex
 	libRoot           string
 	runRoot           string
-	plugins           pluginMap // TODO: figure out why save() doesn't json encode *plugin object
```
Comment (tiborvass) [inline plugin/manager.go]: about the TODO is it still the case?

Comment (anusha-ragunathan) [inline plugin/manager.go]: No. I can do remove that in my next PR.

Comment (tiborvass) [inline plugin/manager.go]: It's already removed, i want to make sure that it's removed on purpose.

**Comment by tiborvass** [comment]
LGTM

**Pull request merged**